packages/experimental/AppearanceAdditions/ios/FRNAppearanceAdditions.m

```diff
@@ -76,18 +80,29 @@ + (BOOL)requiresMainQueueSetup {
     return YES;
 }
 
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(horizontalSizeClass)
-{
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(horizontalSizeClass:(id)reactTag) {
+    RCTUnsafeExecuteOnMainQueueSync(^{
+        if ([reactTag isKindOfClass:[NSNumber class]]) {
+            UIView *view = [[[self bridge] uiManager] viewForReactTag:reactTag];
+            NSString *horizontalSizeClass = RCTHorizontalSizeClassPreference([view traitCollection]);
+            self->_horizontalSizeClass = horizontalSizeClass;
+        }
+    });
     return _horizontalSizeClass;
 }
 
-RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(userInterfaceLevel)
-{
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(userInterfaceLevel:(id)reactTag) {
```

Comment:
Note - I tried overloading this method so that we can call userInterfaceLevel() instead of userInterfaceLevel(null) but ran into issues, it seems like with RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD only one of these methods can be recognized at a time.